util/src/rlp/rlpcompression.rs

```diff
@@ -59,14 +60,14 @@ pub enum RlpType {
 	Snapshot,
 }
 
-fn to_elastic(slice: &[u8]) -> ElasticArray1024<u8> {
-	let mut out = ElasticArray1024::new();
-	out.append_slice(slice);
+fn to_small_vec(slice: &[u8]) -> SmallVec<[u8; 1024]> {
+	let mut out = SmallVec::new();
+	out.extend(slice.to_vec());
```

Comment:
favor `slice.iter().cloned()` here, it'll save a heap allocation in a relatively hot function.